projects/storefrontlib/shared/components/hierarchy/hierarchy/hierarchy.component.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, Input, OnInit, TemplateRef } from '@angular/core';
+import { HierarchyNode } from '../hierarchy-node/hierarchy-node.model';
+import { ActiveCartFacade } from '@spartacus/cart/base/root';
+
+/**
+ * Selector component that displays a tree-based model.
+ */
+@Component({
+  selector: 'cx-hierarchy',
+  templateUrl: './hierarchy.component.html',
+})
+export class HierarchyComponent implements OnInit {
```

Comment:
I see this component has a lot of direct references to the cart library. It seems like its being used specifically for the cart entry groups. If you want to make this component a shared component for use with any library and keep it in `storefrontlib`, you need to remove any reference to the cart and make the component "dumb". 

If you plan to use this component with just the `cart` library, then you should move it to the cart library and rename it to something more targeted (eg. `CartNestedEntryGroupItem` - this example is extremely generalized and should have some more thought put into it).

We can discuss this if you need :)

Reply:
Very good point, I've removed any reference to the `cart` to make it more dumb.